x-pack/plugins/cases/public/components/files/file_type.tsx

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import React from 'react';
+
+import type { FileJSON } from '@kbn/shared-ux-file-types';
+
+import type {
+  ExternalReferenceAttachmentType,
+  ExternalReferenceAttachmentViewProps,
+} from '../../client/attachment_framework/types';
+
+import { AttachmentActionType } from '../../client/attachment_framework/types';
+import { FILE_ATTACHMENT_TYPE } from '../../../common/api';
+import { FileDownloadButtonIcon } from './file_download_button_icon';
+import { FileNameLink } from './file_name_link';
+import { FilePreview } from './file_preview';
+import * as i18n from './translations';
+import { isImage, isValidFileExternalReferenceMetadata } from './utils';
+import { useFilePreview } from './use_file_preview';
+import { FileDeleteButtonIcon } from './file_delete_button_icon';
+
+interface FileAttachmentEventProps {
+  file: FileJSON;
+}
+
+const FileAttachmentEvent = ({ file }: FileAttachmentEventProps) => {
+  const { isPreviewVisible, showPreview, closePreview } = useFilePreview();
+
+  return (
+    <>
+      {i18n.ADDED}
+      <FileNameLink file={file} showPreview={showPreview} />
+      {isPreviewVisible && <FilePreview closePreview={closePreview} selectedFile={file} />}
+    </>
+  );
+};
+
+FileAttachmentEvent.displayName = 'FileAttachmentEvent';
+
+function getFileDownloadButtonIcon(fileId: string) {
+  return <FileDownloadButtonIcon fileId={fileId} />;
+}
+
+function getFileDeleteButtonIcon(caseId: string, fileId: string) {
+  return <FileDeleteButtonIcon caseId={caseId} fileId={fileId} />;
+}
+
+const getFileAttachmentActions = ({ caseId, fileId }: { caseId: string; fileId: string }) => [
```

Comment:
I know I've been MIA for a couple of meetings so I apologize if this was already discussed haha. I believe the mocks have these underneath an overflow menu:

![image](https://user-images.githubusercontent.com/56361221/231587839-8214bd5c-4f2a-4fa9-a9d6-e75aee9e45b2.png)

vs how they are currently in the PR:

![image](https://user-images.githubusercontent.com/56361221/231588107-65ad1634-28ee-41fa-b84c-17be8415f7dd.png)

Reply:
@mdefazio which one do you prefer?

Reply:
Well, I moved those into the ellipsis.